Fail to build PI

I tried to build PI with commit 4546038f5770e84dc0d2bba90f1ee7811c9955df with protobuf==3.6.1 and gRPC v1.17.0
My os is Debian 8, I have python 3.6.9, cmake 3.22.3, make 4.0, gcc 4.9.2
I test gRPC helloworld and it works well.

I use the following command to build:

git clone https://github.com/p4lang/PI.git
git checkout 4546038f5770e84dc0d2bba90f1ee7811c9955df
git submodule update --init --recursive
./autogen.sh
PKG_CONFIG_PATH=/root/bf-sde-9.6.0/install/lib/pkgconfig:$PKG_CONFIG_PATH ./configure CPPFLAGS=-I/root/bf-sde-9.6.0/install/include --prefix=/root/bf-sde-9.6.0/install --with-proto=yes --without-bmv2 --without-internal-rpc --without-cli
make -j8

After make I got the following errors:

In file included from src/digest_mgr.h:24:0,
                 from src/digest_mgr.cpp:22:
./PI/frontends/proto/device_mgr.h:34:24: error: missing binary operator before token "("
 #if __has_cpp_attribute(deprecated)
                        ^
./PI/frontends/proto/device_mgr.h:115:24: error: missing binary operator before token "("
 #if __has_cpp_attribute(deprecated)
                        ^
In file included from src/idle_timeout_buffer.h:25:0,
                 from src/idle_timeout_buffer.cpp:21:
./PI/frontends/proto/device_mgr.h:34:24: error: missing binary operator before token "("
 #if __has_cpp_attribute(deprecated)
                        ^
./PI/frontends/proto/device_mgr.h:115:24: error: missing binary operator before token "("
 #if __has_cpp_attribute(deprecated)
                        ^
In file included from src/device_mgr.cpp:22:0:
./PI/frontends/proto/device_mgr.h:34:24: error: missing binary operator before token "("
 #if __has_cpp_attribute(deprecated)
                        ^
./PI/frontends/proto/device_mgr.h:115:24: error: missing binary operator before token "("
 #if __has_cpp_attribute(deprecated)
                        ^
In file included from src/packet_io_mgr.h:24:0,
                 from src/packet_io_mgr.cpp:21:
./PI/frontends/proto/device_mgr.h:34:24: error: missing binary operator before token "("
 #if __has_cpp_attribute(deprecated)
                        ^
./PI/frontends/proto/device_mgr.h:115:24: error: missing binary operator before token "("
 #if __has_cpp_attribute(deprecated)
                        ^
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -I../../include -I../cpp_out -I../../frontends_extra/cpp -I.. -I../p4info -I../d -Wall -Wextra -g -O2 -std=c++11 -MT src/logging.lo -MD -MP -MF src/.deps/logging.Tpo -c src/logging.cpp -o src/logging.o >/de
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -I../../include -I../cpp_out -I../../frontends_extra/cpp -I.. -I../p4info -I../d -Wall -Wextra -g -O2 -std=c++11 -MT src/table_info_store.lo -MD -MP -MF src/.deps/table_info_store.Tpo -c src/table_info_storinfo_store.o >/dev/null 2>&1
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -I../../include -I../cpp_out -I../../frontends_extra/cpp -I.. -I../p4info -I../d -Wall -Wextra -g -O2 -std=c++11 -MT src/common.lo -MD -MP -MF src/.deps/common.Tpo -c src/common.cpp -o src/common.o >/dev/nu
src/pre_clone_mgr.cpp: In member function 'pi::fe::proto::Status pi::fe::proto::PreCloneMgr::session_set(const CloneSession&, pMgr::GroupId, const SessionTemp&)':
src/pre_clone_mgr.cpp:83:47: warning: missing initializer for member 'pi_clone_session_config_t::direction' [-Wmissing-field-in
   pi_clone_session_config_t session_config = {};  // value-initialization
                                               ^
src/pre_clone_mgr.cpp:83:47: warning: missing initializer for member 'pi_clone_session_config_t::eg_port' [-Wmissing-field-init
src/pre_clone_mgr.cpp:83:47: warning: missing initializer for member 'pi_clone_session_config_t::eg_port_valid' [-Wmissing-fiel
src/pre_clone_mgr.cpp:83:47: warning: missing initializer for member 'pi_clone_session_config_t::mc_grp_id' [-Wmissing-field-in
src/pre_clone_mgr.cpp:83:47: warning: missing initializer for member 'pi_clone_session_config_t::mc_grp_id_valid' [-Wmissing-fi
src/pre_clone_mgr.cpp:83:47: warning: missing initializer for member 'pi_clone_session_config_t::copy_to_cpu' [-Wmissing-field-
src/pre_clone_mgr.cpp:83:47: warning: missing initializer for member 'pi_clone_session_config_t::max_packet_length' [-Wmissing-
src/pre_clone_mgr.cpp:83:47: warning: missing initializer for member 'pi_clone_session_config_t::cos' [-Wmissing-field-initiali
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -I../../include -I../cpp_out -I../../frontends_extra/cpp -I.. -I../p4info -I../d -Wall -Wextra -g -O2 -std=c++11 -MT src/action_helpers.lo -MD -MP -MF src/.deps/action_helpers.Tpo -c src/action_helpers.cpp rs.o >/dev/null 2>&1
Makefile:581: recipe for target 'src/packet_io_mgr.lo' failed
make[4]: *** [src/packet_io_mgr.lo] Error 1
make[4]: *** Waiting for unfinished jobs....
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -I../../include -I../cpp_out -I../../frontends_extra/cpp -I.. -I../p4info -I../d -Wall -Wextra -g -O2 -std=c++11 -MT src/match_key_helpers.lo -MD -MP -MF src/.deps/match_key_helpers.Tpo -c src/match_key_helch_key_helpers.o >/dev/null 2>&1
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -I../../include -I../cpp_out -I../../frontends_extra/cpp -I.. -I../p4info -I../d -Wall -Wextra -g -O2 -std=c++11 -MT src/access_arbitration.lo -MD -MP -MF src/.deps/access_arbitration.Tpo -c src/access_arbiaccess_arbitration.o >/dev/null 2>&1
Makefile:581: recipe for target 'src/idle_timeout_buffer.lo' failed
make[4]: *** [src/idle_timeout_buffer.lo] Error 1
Makefile:581: recipe for target 'src/digest_mgr.lo' failed
make[4]: *** [src/digest_mgr.lo] Error 1
Makefile:581: recipe for target 'src/device_mgr.lo' failed
make[4]: *** [src/device_mgr.lo] Error 1
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -I../../include -I../cpp_out -I../../frontends_extra/cpp -I.. -I../p4info -I../d -Wall -Wextra -g -O2 -std=c++11 -MT src/pre_clone_mgr.lo -MD -MP -MF src/.deps/pre_clone_mgr.Tpo -c src/pre_clone_mgr.cpp -o  >/dev/null 2>&1
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -I../../include -I../cpp_out -I../../frontends_extra/cpp -I.. -I../p4info -I../d -Wall -Wextra -g -O2 -std=c++11 -MT src/pre_mc_mgr.lo -MD -MP -MF src/.deps/pre_mc_mgr.Tpo -c src/pre_mc_mgr.cpp -o src/pre_m2>&1
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -I../../include -I../cpp_out -I../../frontends_extra/cpp -I.. -I../p4info -I../d -Wall -Wextra -g -O2 -std=c++11 -MT src/watch_port_enforcer.lo -MD -MP -MF src/.deps/watch_port_enforcer.Tpo -c src/watch_porrc/watch_port_enforcer.o >/dev/null 2>&1
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -I../../include -I../cpp_out -I../../frontends_extra/cpp -I.. -I../p4info -I../d -Wall -Wextra -g -O2 -std=c++11 -MT src/action_prof_mgr.lo -MD -MP -MF src/.deps/action_prof_mgr.Tpo -c src/action_prof_mgr.cof_mgr.o >/dev/null 2>&1
make[4]: Leaving directory '/root/PI/proto/frontend'
Makefile:1706: recipe for target 'all-recursive' failed
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory '/root/PI/proto'
Makefile:701: recipe for target 'all' failed
make[2]: *** [all] Error 2
make[2]: Leaving directory '/root/PI/proto'
Makefile:450: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/root/PI'
Makefile:381: recipe for target 'all' failed
make: *** [all] Error 2

Any idea what's happening and how to solve this? 
Thanks!

--- a/README.md
+++ b/README.md
@@ -7,6 +7,18 @@
 
 ## Dependencies
 
+### Compiler versions
+
+The following compiler versions are supported:
+
+- gcc / g++: >= 5
+- clang / clang++: >= 3.8
+
+The following compiler versions are tested in CI:
+
+- gcc / g++: 9.3
+- clang / clang++: 8, 10
+
 ### Dependencies based on configure flags
 
 Based on the command-line flags you intend on providing to `configure`, you need

